### DISCUSSION TITLE
Responsive carousel with slidesPerView prop without CLS (Comulative Layout Shift)

### DISCUSSION THREAD

**Discussion by DonHuskini** [davidjerleke/embla-carousel#325] [Show and tell]
Hello everyone! I'd like to share my Embla Carousel adaptation to make it fully responsive across different viewports without causing any layout shift. Some other approaches could be for eg using a hook to detect viewport size but it will cause layout shift and UserAgent for SSR but it's not 100% reliable.

Notice that for mobile screens I don't want to show 1 slide per view but 1 and a bit of the next one which makes it a bit trickier to implement.

https://user-images.githubusercontent.com/17390090/173419426-d41773f6-cd33-4175-a263-8628a9ca01b8.mp4

Implementation:
https://gist.github.com/gabrielmlinassi/91624dcb1f8b137efb7de46f2dcfbac3

**Comment by davidjerleke** [comment]
Thanks a lot for sharing this @gabrielmlinassi 🙂.

Would you mind adding the tech stack in the [OP](https://github.com/davidjerleke/embla-carousel/discussions/325#discussion-4141476) so users interested in this sees that its based on Styled Components etc?

Much appreciated 👍.
